sourcecred/sourcecred
# Dependency cred support for CredEquate

# Description
This adds support for the Dependencies feature (sometimes colloquially called tithing) to CredEquate by adding a new Dependencies auxiliary API. I chose to make this a separate API to take advantage of the CredGrainView as an input instead of raw contributions. The CredGrainView is not calculated inside the API to maintain the Iterable architecture to make future memory optimizations easy.


# Test Plan
1. yarn build
2. checkout sourcecred/cred/gh-pages
3. add the below dependencies.json
3. `scdev credequate` and verify that sourcecred and deepwork are in the cred scores in the console
4. `scdev hybrid` and verify the same for all 3 tables

```
[
    {
        "autoActivateOnIdentityCreation": true,
        "id": "Ec60d6PWymrN0ylmsZOHkg",
        "name": "sourcecred",
        "periods": [
            {
                "startTime": "2019-02-13",
                "weight": 0.1
            }
        ]
    },
    {
        "autoActivateOnIdentityCreation": true,
        "id": "vKNDrdz6Ue8tRCmpO2uAZQ",
        "name": "DeepWork",
        "periods": [
            {
                "startTime": "2019-02-13",
                "weight": 0.2
            }
        ]
    }
]
```
